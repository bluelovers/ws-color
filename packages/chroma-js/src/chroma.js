"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.chroma = void 0;
const Color_1 = __importDefault(require("./Color"));
// @ts-ignore
exports.chroma = (...args) => {
    return new exports.chroma.Color(...args);
};
exports.chroma.Color = Color_1.default;
exports.chroma.version = '@@version';
exports.chroma.chroma = exports.chroma;
exports.chroma.default = exports.chroma;
Object.defineProperty(exports.chroma, "__esModule", { value: true });
exports.default = exports.chroma;
//# sourceMappingURL=chroma.js.map