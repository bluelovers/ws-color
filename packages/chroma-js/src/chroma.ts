import Color from './Color';
import { IChromaConstructor } from './types';

export interface chroma extends IChromaConstructor
{
	(...args): Color,
	Color: typeof Color;
	version: string;
	chroma: chroma,
	default: chroma,
}

// @ts-ignore
export const chroma: chroma = (...args) =>
{
	return new chroma.Color(...args);
};

chroma.Color = Color;
chroma.version = '@@version'

chroma.chroma = chroma;
chroma.default = chroma;

Object.defineProperty(chroma, "__esModule", { value: true });

export default chroma;
