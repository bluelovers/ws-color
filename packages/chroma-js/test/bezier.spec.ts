import chroma from '..';

describe(`simple two color linear interpolation`, () =>
{

	let fn = chroma.bezier(['white', chroma('black').alpha(0)]);

	test(`starts from white`, () =>
	{

		let actual = fn(0);

		expect(actual.hex()).toStrictEqual('#ffffff');

		expect(actual).toMatchSnapshot();

	});

	test(`ends in transparent black`, () =>
	{

		let actual = fn(1);

		expect(actual.hex()).toStrictEqual('#00000000');

		expect(actual).toMatchSnapshot();

	});

	test(`center is transluscent grey`, () =>
	{

		let actual = fn(0.5);

		expect(actual.hex()).toStrictEqual('#77777780');

		expect(actual).toMatchSnapshot();

	});

})

describe(`three color quadratic bezier interpolation`, () =>
{

	let fn = chroma.bezier(['white', chroma('red').alpha(.5), chroma('black').alpha(0)]);

	test(`starts from white`, () =>
	{

		let actual = fn(0);

		expect(actual.hex()).toStrictEqual('#ffffff');

		expect(actual).toMatchSnapshot();

	});

	test(`ends in transparent black`, () =>
	{

		let actual = fn(1);

		expect(actual.hex()).toStrictEqual('#00000000');

		expect(actual).toMatchSnapshot();

	});

	test(`center is a transluscent greyish red`, () =>
	{

		let actual = fn(0.5);

		expect(actual.hex()).toStrictEqual('#c45c4480');

		expect(actual).toMatchSnapshot();

	});

})

describe(`four color cubic bezier interpolation`, () =>
{

	let fn = chroma.bezier([
		'white',
		chroma('yellow').alpha(1 / 3),
		chroma('red').alpha(2 / 3),
		chroma('black').alpha(0),
	]);

	test(`starts from white`, () =>
	{

		let actual = fn(0);

		expect(actual.hex()).toStrictEqual('#ffffff');

		expect(actual).toMatchSnapshot();

	});

	test(`ends in transparent black`, () =>
	{

		let actual = fn(1);

		expect(actual.hex()).toStrictEqual('#00000000');

		expect(actual).toMatchSnapshot();

	});

	test(`1st quarter`, () =>
	{

		let actual = fn(0.25);

		expect(actual.hex()).toStrictEqual('#ffe085a7');

		expect(actual).toMatchSnapshot();

	});

	test(`center`, () =>
	{

		let actual = fn(0.5);

		expect(actual.hex()).toStrictEqual('#e6973580');

		expect(actual).toMatchSnapshot();

	});

	test(`3rd quarter`, () =>
	{

		let actual = fn(0.75);

		expect(actual.hex()).toStrictEqual('#91421358');

		expect(actual).toMatchSnapshot();

	});

})

describe(`five color diverging quadratic bezier interpolation`, () =>
{

	let fn = chroma.bezier([
		'darkred',
		chroma('orange').alpha(.75),
		chroma('snow').alpha(.5),
		chroma('lightgreen').alpha(.25),
		chroma('royalblue').alpha(0),
	]);

	test(`starts from darkred`, () =>
	{

		let actual = fn(0);

		expect(actual.hex()).toStrictEqual('#8b0000');

		expect(actual).toMatchSnapshot();

	});

	test(`ends in transparent royalblue`, () =>
	{

		let actual = fn(1);

		expect(actual.hex()).toStrictEqual('#4169e100');

		expect(actual).toMatchSnapshot();

	});

	test(`1st quarter`, () =>
	{

		let actual = fn(0.25);

		expect(actual.hex()).toStrictEqual('#e9954ebf');

		expect(actual).toMatchSnapshot();

	});

	test(`center is snow`, () =>
	{

		let actual = fn(0.5);

		expect(actual.hex()).toStrictEqual('#fffafa80');

		expect(actual).toMatchSnapshot();

	});

	test(`3rd quarter`, () =>
	{

		let actual = fn(0.75);

		expect(actual.hex()).toStrictEqual('#a6cfc140');

		expect(actual).toMatchSnapshot();

	});

})

