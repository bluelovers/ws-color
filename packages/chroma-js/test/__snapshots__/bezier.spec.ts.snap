// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`five color diverging quadratic bezier interpolation 1st quarter 1`] = `
Color {
  "_rgb": Array [
    233.19030009653343,
    149.48568936490597,
    77.70818925383125,
    0.75,
  ],
}
`;

exports[`five color diverging quadratic bezier interpolation 3rd quarter 1`] = `
Color {
  "_rgb": Array [
    165.95197295723432,
    206.80328083106684,
    193.23203906355687,
    0.25,
  ],
}
`;

exports[`five color diverging quadratic bezier interpolation center is snow 1`] = `
Color {
  "_rgb": Array [
    254.9999889601456,
    250.0000173961273,
    249.99999732944465,
    0.5,
  ],
}
`;

exports[`five color diverging quadratic bezier interpolation ends in transparent royalblue 1`] = `
Color {
  "_rgb": Array [
    65.00002095351515,
    104.99999975559058,
    225.00000209416496,
    0,
  ],
}
`;

exports[`five color diverging quadratic bezier interpolation starts from darkred 1`] = `
Color {
  "_rgb": Array [
    138.9999882021398,
    0.00011212162605855155,
    0,
    1,
  ],
}
`;

exports[`four color cubic bezier interpolation 1st quarter 1`] = `
Color {
  "_rgb": Array [
    255,
    224.03899248147033,
    132.81503252512528,
    0.65625,
  ],
}
`;

exports[`four color cubic bezier interpolation 3rd quarter 1`] = `
Color {
  "_rgb": Array [
    144.87099553702365,
    65.651166141413,
    18.759023083692984,
    0.34375,
  ],
}
`;

exports[`four color cubic bezier interpolation center 1`] = `
Color {
  "_rgb": Array [
    230.1559171578932,
    150.8554374112126,
    52.84422558429743,
    0.5,
  ],
}
`;

exports[`four color cubic bezier interpolation ends in transparent black 1`] = `
Color {
  "_rgb": Array [
    2.0424926888958734e-7,
    2.042492495365618e-7,
    2.0424925616526716e-7,
    0,
  ],
}
`;

exports[`four color cubic bezier interpolation starts from white 1`] = `
Color {
  "_rgb": Array [
    254.99998940693285,
    255,
    254.99999736707582,
    1,
  ],
}
`;

exports[`simple two color linear interpolation center is transluscent grey 1`] = `
Color {
  "_rgb": Array [
    118.91328116479764,
    118.91329253054577,
    118.9132842032307,
    0.5,
  ],
}
`;

exports[`simple two color linear interpolation ends in transparent black 1`] = `
Color {
  "_rgb": Array [
    2.0424926888958734e-7,
    2.042492495365618e-7,
    2.0424925616526716e-7,
    0,
  ],
}
`;

exports[`simple two color linear interpolation starts from white 1`] = `
Color {
  "_rgb": Array [
    254.99998940693285,
    255,
    254.99999736707582,
    1,
  ],
}
`;

exports[`three color quadratic bezier interpolation center is a transluscent greyish red 1`] = `
Color {
  "_rgb": Array [
    196.38575706795183,
    91.71313890151566,
    67.65992964943767,
    0.5,
  ],
}
`;

exports[`three color quadratic bezier interpolation ends in transparent black 1`] = `
Color {
  "_rgb": Array [
    2.0424926888958734e-7,
    2.042492495365618e-7,
    2.0424925616526716e-7,
    0,
  ],
}
`;

exports[`three color quadratic bezier interpolation starts from white 1`] = `
Color {
  "_rgb": Array [
    254.99998940693285,
    255,
    254.99999736707582,
    1,
  ],
}
`;
